### ISSUE TITLE
Build a demo for Style Transfer

### ISSUE THREAD

**Issue by dexter1691** [Cloud-CV/Origami#30]
Build a demo for the fun application of style transfer from one image to another by writing a wrapper around this code:
~https://github.com/cysmith/neural-style-tf~

EDIT (@abhshkdz): Let's build a demo for this implementation instead — https://github.com/tensorflow/magenta/tree/master/magenta/models/image_stylization, it's faster than the original style transfer algorithm, allows for multiple styles to be applied in a single feedforward step using a single network via a change in the instance normalization parameters, and the code (being part of the official repo) is well-maintained.

**Comment by souravsingh** [comment]
@dexter1691 I am interested in working on this. I have to create a wrapper for the above mentioned library and add it to cvfy-lib, correct?

**Comment by deshraj** [comment]
@souravsingh Yes, you are right. Let us know if you run into issues with the cvfy package. :)

**Comment by tocttou** [comment]
>  create a wrapper for the above mentioned library and add it to cvfy-lib, correct?

"Build a demo" seems to suggest that that the demo is to be added to cvfy-frontend (maybe the instance running it http://cvfy.cloudcv.org), and not cvfy-lib (which is the python lib) if I am reading it correctly.

**Comment by souravsingh** [comment]
@tocttou We would have to add the ML evaluation code before we can build a demo, correct?

**Comment by tocttou** [comment]
A demo is built in two parts:

1. You _use_ cvfy-lib to integrate that style transfer demo with a webserver.
2. You link that webserver to the cvfy-frontend application instance and share the generated demo link.

I suggest going through the following links:  
1. http://cvfy.cloudcv.org/libdocs/configuration [for using cvfy-lib]
2. http://cvfy.cloudcv.org/gettingstarted/configuration [for creating a demo by linking the above to cvfy-frontend]

**Comment by tocttou** [comment]
Note that creating a demo does not require making changes to cvfy-lib. You just need to import it. Maybe we need some example code snippets to showcase this.

You can take a look at this simple color-to-grayscale app made using functions imported from cvfy-lib: https://github.com/tocttou/samplecv/blob/master/app.py

Also read the docs once.

**Comment by anweshknayak** [comment]
@tocttou Are there any demo for problems like this? I saw one video for configuring any app.

**Comment by tocttou** [comment]
Functional public demo: not yet. That is the task, no? :p

**Comment by dexter1691** [comment]
Hehe. @tocttou is right! The code should be fairly straight forward to read the cvfy-lib code to see what needs to be done. You just need to use it as a wrapper to take input and send output to cvfy-frontend.

**Comment by harvenomey** [comment]
I tried implementing magenta's style transfer but I'm stuck on an issue.
https://github.com/tensorflow/magenta/issues/530

While it is fixed, can I move on further to implement fast-style-transfer?
https://github.com/lengstrom/fast-style-transfer

**Comment by nikhilweee** [comment]
I tried implementing magenta's style transfer but I'm stuck on an issue.
https://github.com/tensorflow/magenta/issues/530

While it is fixed, can I move on further to implement fast-style-transfer?
https://github.com/lengstrom/fast-style-transfer

**Comment by dexter1691** [comment]
@nikhilweee : 530 suggets that the fix is already in the master! Can you try install the master in another virtual env and see if that works? 
Also implementing fast style transfer sounds good too.

**Comment by souravsingh** [comment]
@dexter1691 Would it be fine if the app is built on PyTorch?

**Comment by harvenomey** [comment]
@dexter1691 : I'm already on the latest master. I have the fix (I've cross-checked), but it still doesn't work.  Anyways I'll try doing fast-style-transfer in the meantime.

**Comment by nikhilweee** [comment]
@dexter1691 I already have the fix (I've cross checked) but I'm still facing the issue. Anyways I'll try implementing fast-style-transfer in the meanwhile.

**Comment by nikhilweee** [comment]
Hey @dexter1691 : I've completed the style transfer demo. It's at https://github.com/nikhilweee/fast-style-transfer. Here is a relevant screenshot:
![Input](https://cloud.githubusercontent.com/assets/8774119/23831397/3d866a78-0746-11e7-9332-f3b6fa38c1db.png)
![Output](https://cloud.githubusercontent.com/assets/8774119/23831400/4ae40b76-0746-11e7-8cf2-e7ba0ad3355c.png)

**Comment by dexter1691** [comment]
Can you create a docker image for this and share it with us? We will then add the demo on the wiki and may be host it on CloudCV as well.

**Comment by nikhilweee** [comment]
Sure. I'm working on it.

**Comment by deshraj** [comment]
@nikhilweee let us know whenever you are done with this so that we can host it on CloudCV server.

**Comment by KrUciFieR-Jr** [comment]
Is this issue open? if yes then I would like to work on it

**Comment by nikhilweee** [comment]
I'm sorry I haven't been able to work on this one. Could someone create the docker image form https://github.com/nikhilweee/fast-style-transfer ?

**Comment by KrUciFieR-Jr** [comment]
I will help you to create the docker image.
